pkg/operators/cli/clioperator.go

```diff
@@ -179,6 +180,36 @@ func (o *cliOperatorInstance) ExtraParams(gadgetCtx operators.GadgetContext) api
 	return api.Params{fields, mode}
 }
 
+func parseFields(fieldsString string, defaultFields []string) []string {
```

Comment:
There are some corner cases that aren't handled by this implementation:

```bash 
# It should not fail:
$ sudo ig run trace_exec:latest --fields=-timestamp
Error: starting operators: pre-starting operator "cli": setting fields: column "" is invalid

# it should print timestamp and error 
$ sudo ig run trace_exec:latest --fields=+timestamp,error
                                                                                                                                   ERROR
```

What do you think about this alternative to handle them :

```go
func parseFields(fieldsString string, defaultFields []string) []string {
	fields := strings.Split(fieldsString, ",")

	addedFields := []string{}
	deletedFields := []string{}
	explicitFields := []string{}

	for _, field := range fields {
		field = strings.TrimSpace(field)
		if field == "" {
			continue
		}
		switch field[0] {
		case '+':
			addedFields = append(addedFields, field[1:])
		case '-':
			deletedFields = append(deletedFields, field[1:])
		default:
			if !slices.Contains(explicitFields, field) {
				explicitFields = append(explicitFields, field)
			}
		}
	}

	result := defaultFields
	if len(explicitFields) > 0 {
		result = explicitFields
	}

	for _, field := range addedFields {
		if !slices.Contains(result, field) {
			result = append(result, field)
		}
	}

	for _, field := range deletedFields {
		result = slices.DeleteFunc(result, func(s string) bool { return s == field })
	}

	return result
}
```

Reply:
Yes @mauriciovasquezbernal this is little different approach and looks really good to me.  
```
addedFields := make([]string, 0, len(fields))
	deletedFields := make([]string, 0, len(fields))
	explicitFields := make([]string, 0, len(fields))
```
just doing this little change `Preallocate slices` as this can improve performance slightly by reducing the number of memory allocations.

I was also thinking of using trimming space, but its not that much required. 
its working fine now.
```bash
$ sudo -E ./ig run trace_exec:latest --fields=+timestamp,error 
[sudo] password for h4logen: 
INFO[0000] Experimental features enabled                
WARN[0000] Ignoring runtime "cri-o" with non-existent socketPath "/run/crio/crio.sock" 
ERROR                              TIMESTAMP 

```
Please tell me is this looks good to you? then I will change documentation regarding this. Thank you

Reply:
> just doing this little change Preallocate slices as this can improve performance slightly by reducing the number of memory allocations.

I'm not sure about this since you're preallocating all three slices to have the same size of the input fields, i.e. you're preallocating more space than will be ever used. 
Also, this code is not part of a hot path, it's only run before running the gadget and the number of fields are rather small  (<100), so I don't think performance here is too critical. 

In any case, I don't have a strong opinion on the approach, any is good to me.

Reply:
> Please tell me is this looks good to you? then I will change documentation regarding this. Thank you

Yes, it LGTM. thanks! 

btw. please format your commit message:
- title: add a "clioperator:" prefix 
- body: add a description of **what** is being done, something similar to the PR description

Reply:
@mauriciovasquezbernal I've changed commit message.

Reply:
I still see the commit message as 

```
Signed-off-by: h4l0gen <ks3913688@gmail.com>

using addedfield and hidefield map logic

Signed-off-by: h4l0gen <ks3913688@gmail.com>

changing back to slices instead of map

Signed-off-by: h4l0gen <ks3913688@gmail.com>

including corner cases left

Signed-off-by: h4l0gen <ks3913688@gmail.com>

including corner cases left

Signed-off-by: h4l0gen <ks3913688@gmail.com>
```

Would you mind fixing it to have:
- a short description of the change in this commit
- your signoff 

(you can use `git commit --amend` to edit it)

Reply:
Hi, you put all the content on the title. Please split that into title + body. 

Also, please don't forget [this](https://github.com/inspektor-gadget/inspektor-gadget/pull/3284#issuecomment-2296303301) comment about adding documentation.

Reply:
@mauriciovasquezbernal added documentation, please review it.

Reply:
looks good. Can you also add the kubectl-gadget one to keep it uniform?

Reply:
@mauriciovasquezbernal made changes, we are good to go now, Please review.